packages/retail-ui-extensions/src/components/shared/CoreInputField.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Represents an action type for the text field components.
+ * @typedef {Object} InputAction
+ * @property {string} label - The text displayed in the button.
+ * @property {boolean} [disabled] - Whether the button can be pressed.
+ * @property {function(): void} onPress - A callback to be performed.
+ */
+export interface InputAction {
+  label: string;
+  onPress: () => void;
+  disabled?: boolean;
+}
+
+/**
+ * Represents the properties for the text fied components.
```

Comment:
What are you guys thoughts on adding these type of comments to our package components? Do you think this is beneficial for developers or possibly too much overhead to update if front facing documentation changes?

Reply:
Beneficial! I always appreciate when I can click through my IDE to see these sorts of comments.